### PR TITLE
Dart2 fixes for bazel_codegen

### PR DESCRIPTION
**Pull request by jakemac53**
All tests now pass, will validate internally now.

cc @keertip

### CODE REVIEW

**Comment by jakemac53** [comment]
Fixes https://github.com/dart-lang/build/issues/1304

**Comment by jakemac53** [comment]
Looks like some more fixes are needed to get things truly working internally, will keep iterating on this.

File: bazel_codegen/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# 0.3.1+2
+
+- Added support for running in Dart 2 mode via `--preview-dart-2`.
```
Comment (natebosch) [inline bazel_codegen/CHANGELOG.md]: I'd phrase this differently. "Fixed runtime errors with running in Dart 2 mode via `--preview-dart-2`."

Comment (jakemac53) [reply bazel_codegen/CHANGELOG.md]: done

**Pull request merged**